build/tools/CFamilyJarPreProcessor/Common.cs

```diff
@@ -0,0 +1,251 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using SharpCompress.Compressors.Xz;
+
+namespace CFamilyJarPreProcessor
+{
+    internal static class Common
+    {
+        /// <summary>
+        /// Returns the full path to the local directory in which the plugin will be cached at build time
+        /// </summary>
+        /// <param name="pluginFolderName">Per-plugin unique folder name</param>
+        /// <remarks>By default the plugins will be stored under the user's %LocalAppData% folder e.g. C:\Users\JoeBloggs\AppData\Local.
+        /// An alternative root directory can be specified by setting the environment variable SONARLINT_INTERNAL_PLUGIN_CACHE_DIR.
+        /// This might be necessary if the user name is long so the full paths of the files being unpacked under the root folder
+        /// exceed the maximum path length.</remarks>
+        public static string GetLocalBuildTimePluginCacheDir(string pluginFolderName)
```

Comment:
Copied from the existing MSBuild task.
I think the original version might have contained non-Common code i.e. code specific to either the CFamily or JS/TS plugin. I can't remember what I did about that, if anything.